AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java

```diff
@@ -2124,7 +2124,7 @@ protected void displayCardQuestion(boolean reload) {
         // If the user wants to show the answer automatically
         if (mUseTimer) {
             long delay = mWaitAnswerSecond * 1000 + mUseTimerDynamicMS;
-            if (delay > 0) {
+            if (mWaitAnswerSecond > 0 && delay > 0) {
```

Comment:
Okay - this is interesting - so I checked and `mUseTimerDynamicMS` is "the duration of any sound", while `mWaitAnswerSecond` is definitely supposed to be how long to wait for auto-advance, with `0` confirmed (via examination of previous logic) to mean "no auto-advance".

So you are correctly that before this change if there is audio, we will auto-advance even if auto-advance is otherwise set to zero

After this change, the error is fixed, however I think the *most* correct fix is not to check delay at all, it is even simpler - I think if we just check whether we should auto-advance or not, that' sufficient, but then if we determine we should, obviously we want to postDelayed for the full delay

So I think this is most correct. What do you think?

```suggestion
            if (mWaitAnswerSecond > 0) {  // a wait of zero means auto-advance is disabled
```